editoast/src/models/paced_train.rs

```diff
@@ -57,6 +57,58 @@ pub struct PacedTrain {
 }
 
 impl PacedTrain {
+    pub fn apply_exceptions(&self, exceptions: &[PacedTrainException]) -> Vec<TrainSchedule> {
+        exceptions
+            .iter()
+            .map(|exception| self.apply_exception(exception))
+            .collect()
+    }
+
+    pub fn apply_exception(&self, exception: &PacedTrainException) -> TrainSchedule {
+        let mut train_schedule = Self::into_first_occurrence(self.clone());
+
+        if let Some(change_groupe) = &exception.train_name {
+            train_schedule.train_name = change_groupe.value.clone();
+        }
+        if let Some(change_groupe) = &exception.rolling_stock {
+            train_schedule.comfort = change_groupe.comfort;
+            train_schedule.rolling_stock_name = change_groupe.rolling_stock_name.clone();
+        }
+        if let Some(change_groupe) = &exception.rolling_stock_category {
+            train_schedule.category = change_groupe.value.clone().map(TrainCategory);
+        }
+        if let Some(change_groupe) = &exception.labels {
+            train_schedule.labels = change_groupe
+                .value
+                .iter()
+                .map(|label| Some(label.clone()))
```

Comment:
nitpick: I believe you can probably do the following.

```suggestion
                .iter()
                .cloned()
                .map(Some)
```